lib/datadog/appsec/contrib/active_record/patcher.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require_relative '../patcher'
+require_relative 'instrumentation'
+
+module Datadog
+  module AppSec
+    module Contrib
+      module ActiveRecord
+        # AppSec patcher module for ActiveRecord
+        module Patcher
+          include Datadog::AppSec::Contrib::Patcher
+
+          module_function
+
+          def patched?
+            Patcher.instance_variable_get(:@patched)
+          end
+
+          def target_version
+            Integration.version
+          end
+
+          def patch
+            ActiveSupport.on_load :active_record do
+              if defined? ::ActiveRecord::ConnectionAdapters::SQLite3Adapter
+                ::ActiveRecord::ConnectionAdapters::SQLite3Adapter.prepend(Patcher.prepended_class_name(:sqlite3))
+              end
+
+              if defined? ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter
+                ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(Patcher.prepended_class_name(:postgresql))
+              end
+
+              if defined? ::ActiveRecord::ConnectionAdapters::Mysql2Adapter
+                ::ActiveRecord::ConnectionAdapters::Mysql2Adapter.prepend(Patcher.prepended_class_name(:mysql2))
+              end
+            end
+          end
+
+          def prepended_class_name(adapter_name)
+            if ::ActiveRecord.gem_version >= Gem::Version.new('7.1')
+              Instrumentation::InternalExecQueryAdapterPatch
+            elsif adapter_name == :postgresql && !defined?(::ActiveRecord::ConnectionAdapters::JdbcAdapter)
+              Instrumentation::ExecuteAndClearAdapterPatch
+            else
+              Instrumentation::ExecQueryAdapterPatch
+            end
+          end
```

Comment:
I think this method creates confusion. We need an adapter name only for `postgres`, for the rest simple version check is working.

What if we extract the version check logic into just normal assignment at the top of the `patch` method

```ruby
def patch
  instrumentation_module = if ::ActiveRecord.gem_version >= Gem::Version.new('7.1')
    Instrumentation::InternalExecQueryAdapterPatch
  else
    Instrumentation::ExecQueryAdapterPatch
  end
  
  # ... omitted ...
end
```

and then for `postgres` we can use additional check right in place

```ruby
if defined?(::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter)
 instrumentation_module = Instrumentation::ExecuteAndClearAdapterPatch unless defined?(::ActiveRecord::ConnectionAdapters::JdbcAdapter)
 ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(instrumentation_module)
end
```

Reply:
it's a bit more complicated - we need this adapter only for postgres, but for ActiveRecord versions < 7.1 and not for JRuby